### ISSUE TITLE
Minecraft /timings report crash server

### ISSUE THREAD

**Issue by Draghonir** [oshi/oshi#2312]
<img width="635" alt="219141791-7e804e58-63f4-4b18-ab15-fbe85ce88aeb" src="https://user-images.githubusercontent.com/107469643/219449723-7cb81821-b9df-4e98-994b-ba02cd1e1a6d.png">

with the commands /timings report the server crash and restart in the log oshi doing crash all.

 https://pastes.dev/C7JDdyAXEc

**Comment by dbwiddis** [comment]
Thanks for the report.   The line of code in the stack trace is intended to check whether the system has `libudev` loaded, which would be tested via JNA's `Udev` class.  Normally that would generate an `UnsatisfiedLinkError` which is caught for this test.  It would not catch a `ClassDefNotFound` error in 6.2.2.

That code was changed in OSHI 6.3.0 to also catch `ClassDefNotFound`, so upgrading OSHI may appear to fix the problem.

However I think the underlying problem may be an earlier version of JNA brought in via transitive dependency.

It's hard to tell from the existing logs, but at this point I'm not sure there's anything I can do to fix it (that wasn't fixed in 6.3.0).

**Comment by snk33** [comment]
Hello @dbwiddis 

I'm the founder of the GSP (GameServer Provider) involved here :)

We try to find out a way to make Oshi work properly into our sandboxing system. I've tried to look at the source code but I can't understand what's blocking here.

Server is running on Debian 11 with libudev1 installed, I guess it's more related to some settings to change on our sandbox. 

**CPU part**

The sandbox can access both /proc/cpuinfo and /sys/devices/system/cpu/cpuX so I don't get why it cannot work.

**Memory part**

From what I understand reading source code, it may try dmidecode as sudo to work so there's nothing we can do about it (except indeed upgrading the lib to 6.3.0 indeed).
Maybe I missed something there ?

**Comment by dbwiddis** [comment]
Hey @snk33 the `ClassDefNotFound` normally indicates an older version of JNA on the classpath.  My first suggestion would be to use your dependency manager to make sure you have the latest `jna` and `jna-platform` dependencies pulled in.  Other than that there's not much I can debug from the provided info.

**Comment by snk33** [comment]
It's used in Paper Minecraft server version.

From what I see, the builds are :
- oshi-core 6.2.2
- jna-platform 5.12.1

It looks up-to-date, no ?

From what I see they don't plan to update dependencies like oshi-core soon : https://github.com/PaperMC/Paper/issues/8857

**Comment by dbwiddis** [comment]
> From what I see they don't plan to update dependencies like oshi-core soon : [PaperMC/Paper#8857](https://github.com/PaperMC/Paper/issues/8857)

It's reasonable not to update something that's not broken. I'm sure they would update if you could demonstrate a failing test case/configuration that an update would fix.  Unfortunately as I noted in my first response, while an OSHI update would avoid a crash with a more graceful failure, it wouldn't fix the underlying problem.

> From what I see, the builds are :
> 
> * oshi-core 6.2.2
> * jna-platform 5.12.1
> 
> It looks up-to-date, no ?

That assumes those are the actual versions being imported.  True for OSHI as shown by the stack trace, but possibly not for JNA.

The relevant lines in the stack trace:
```
java.lang.NoClassDefFoundError: Could not initialize class oshi.software.os.linux.LinuxOperatingSystem
	at oshi.hardware.platform.linux.LinuxCentralProcessor.initProcessorCounts(LinuxCentralProcessor.java:171) ~[oshi-core-6.2.2.jar:6.2.2]
```
That's this line:
https://github.com/oshi/oshi/blob/5a2d022ec872294fb6e63434f3e842435bb68359/oshi-core/src/main/java/oshi/hardware/platform/linux/LinuxCentralProcessor.java#L171-L173

where `HAS_UDEV` is declared like this:
https://github.com/oshi/oshi/blob/5a2d022ec872294fb6e63434f3e842435bb68359/oshi-core/src/main/java/oshi/software/os/linux/LinuxOperatingSystem.java#L89-L98

So the error indicates Java cannot find this import:
```java
import com.sun.jna.platform.linux.Udev;
```

That class was added to JNA in version 5.6.0.

**TLDR: There is a version of `jna-platform` on the classpath of 5.5.0 or earlier.**

And there's nothing I can do here at OSHI to fix that. This needs to be fixed up in the dependency resolution of whatever project is importing OSHI and JNA.

The only place I see `jna-platform` in the PaperMC project is [this line](https://github.com/PaperMC/Paper/blob/155aa36d89b260ef5841615899299756b5983c0a/patches/server/0001-Setup-Gradle-project.patch#L363) which is part of a diff _removing_ it from somewhere.  OSHI is "removed" in that same diff.  So clearly there's some script somewhere creating the necessary files before even starting to track this down.

Looking at the instructions [here](https://github.com/PaperMC/Paper#how-to-compiling-jar-from-source) I cloned the project and ran `gradle applyPatches`.  The dependencies look fine here, noting that the JNA dependency is transitive from OSHI, so any other dependency higher up on the dependency tree might override it.
```
     +--- com.github.oshi:oshi-core:6.2.2
     |    +--- net.java.dev.jna:jna:5.12.1
     |    +--- net.java.dev.jna:jna-platform:5.12.1
     |    |    \--- net.java.dev.jna:jna:5.12.1
     |    \--- org.slf4j:slf4j-api:1.7.36 -> 2.0.1
```

The original screen capture at the top of this issue shows lots of plugins being loaded.  I suspect one of those plugins may be bringing JNA into the classpath.

If PaperMC is the top level program you are running, it may make sense to request they update their configuration to explicitly match the JNA version to OSHI version at a higher level on the dependency tree, or import OSHI's dependencyManagement field from the POM.  Alternately you can look at each of the plugins and see which one is bringing in an older JNA.

**Comment by dbwiddis** [comment]
So I realized looking at this code that the line `Udev udev = null` is outside of the try/catch and is probably the point where the incorrect import is happening.  So even the catch in 6.3.0 wouldn't fix this. :|
